packages/frontend/src/components/DashboardTiles/TileBase/index.tsx

```diff
@@ -192,18 +193,29 @@ const TileBase = <T extends Dashboard['tiles'][number]>({
                     <ChartContainer className="non-draggable sentry-block fs-block cohere-block">
                         {children}
                     </ChartContainer>
-
-                    <TileUpdateModal
-                        className="non-draggable"
-                        isOpen={isReplacingChart}
-                        tile={tile}
-                        onClose={() => setIsReplacingChart(false)}
-                        onConfirm={(data) => {
-                            onEdit(data);
-                            setIsReplacingChart(false);
-                        }}
-                    />
-
+                    {tile.type === DashboardTileTypes.SAVED_CHART ? (
+                        <ChartUpdateModal
+                            className="non-draggable"
+                            opened={isReplacingChart}
+                            tile={tile}
+                            onClose={() => setIsReplacingChart(false)}
+                            onConfirm={(newChartTile) => {
+                                onEdit(newChartTile);
+                                setIsReplacingChart(false);
+                            }}
+                        />
+                    ) : (
+                        <TileUpdateModal
+                            className="non-draggable"
+                            isOpen={isReplacingChart}
+                            tile={tile}
+                            onClose={() => setIsReplacingChart(false)}
+                            onConfirm={(newTile) => {
+                                onEdit(newTile);
+                                setIsReplacingChart(false);
+                            }}
+                        />
+                    )}
                     <TileUpdateChartTitle
```

Comment:
@rephus I noticed the title is not updating when I update a tile 🤔  any thoughts on how to fix this?

Reply:
mm, the title is passed as a property on DashboardChartTileMain and TileBase, if the chart is different, the title should be different too, :thinking: